app/src/common/shared/org/mozilla/vrbrowser/browser/engine/EngineProvider.kt

```diff
@@ -25,7 +25,11 @@ object EngineProvider {
 
             builder.crashHandler(CrashReporterService::class.java)
             builder.contentBlocking(ContentBlocking.Settings.Builder()
-                    .antiTracking(ContentBlocking.AntiTracking.AD or ContentBlocking.AntiTracking.SOCIAL or ContentBlocking.AntiTracking.ANALYTIC)
+                    .antiTracking(
+                            ContentBlocking.AntiTracking.AD or
+                            ContentBlocking.AntiTracking.SOCIAL or
+                            ContentBlocking.AntiTracking.ANALYTIC)
+                    .enhancedTrackingProtectionLevel(SettingsStore.getInstance(context).trackingProtectionLevel)
```

Comment:
I think we only want one or the other set at one time?

Reply:
These are independent, actually. `antiTracking` controls what *content* gets blocked when `GeckoSessionSettings.useTrackingProtection` is true, while `enhancedTrackingProtectionLevel` controls what *cookies* get blocked when our cookie behavior is `ACCEPT_NON_TRACKERS`.

Reply:
The way I understood what was explained to me is that we only want one active at a time. If we are blocking cookies we don't want to be blocking content as that is the purpose of ETP, it just blocks the tracking cookies and the content is all loaded which reduce breakage, where content blocking is more strict but also causes more breakage.

Reply:
> These are independent, actually. `antiTracking` controls what _content_ gets blocked when `GeckoSessionSettings.useTrackingProtection` is true, while `enhancedTrackingProtectionLevel` controls what _cookies_ get blocked when our cookie behavior is `ACCEPT_NON_TRACKERS`.

@droeh I guess I'm still confused, I don't see `GeckoSessionSettings.useTrackingProtection()` except when the session is created. Does this mean it never gets turned off?

Reply:
Sorry for the slow reponse, I'm not used to getting work mail from github!
There are cases where you might want to leave ETP and TP both enabled, for example you may want to enable TP only for one category and ETP for the rest. In general they are only exposed in UI mutually exclusively, though (eg as "Standard" and "Strict" in desktop).

And yeah, I missed that, but you definitely need to update `GeckoSessionSettings.useTrackingProtection()` to control TP, and do it for every session. (Yes, that sucks, it's on the list of things to change in the API.)

I'll keep an eye on github mail and respond quicker if you have any more questions.